registry/consul/backend.go

```diff
@@ -36,26 +37,47 @@ func NewBackend(cfg *config.Consul) (registry.Backend, error) {
 	return &be{c: c, dc: dc, cfg: cfg}, nil
 }
 
-func (b *be) Register() error {
-	if !b.cfg.Register {
+func (b *be) Register(serviceName string) error {
```

Comment:
Then here you first unregister names that no longer exist and register new names.